### ISSUE TITLE
admin-cli invalid credentials

### ISSUE THREAD

**Issue by taylorfssaraiva** [keycloak/keycloak#30866]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/cli

### Describe the bug

When trying to authenticate `kcadm.sh` with `admin` password, an invalid credential error is returned. 

When running the command bellow,  we're getting `Invalid user credentials [invalid_grant]` 
`kcadm.sh config credentials --server "$KEYCLOAK_SERVER_URL" --realm "master" --user "$KEYCLOAK_ADMIN" --password "$KEYCLOAK_ADMIN_PASSWORD"`

We tried using the KC_CLI_PASSWORD variable introduced in the latest version. It doesn't return any error for credentials, but it asks for client-secret when running any other kcadm command. But `admin-cli` doesn't have any secret.
```
export KC_CLI_PASSWORD=$KEYCLOAK_ADMIN_PASSWORD
kcadm.sh config credentials --server "$KEYCLOAK_SERVER_URL" --realm "master" --user "$KEYCLOAK_ADMIN"
kcadm.sh get users
```
<img width="715" alt="Screenshot 2024-06-27 at 11 56 36" src="https://github.com/keycloak/keycloak/assets/4632000/a17d9ef6-f299-4203-a87b-0931afb92f72">

### Version

25.0.1

### Regression

- [ ] The issue is a regression

### Expected behavior

kcadm.sh is authenticated and the following commands doesn't ask for authentication 

### Actual behavior

Invalid credentials

### How to Reproduce?

`docker run -p 8080:8080 -e KEYCLOAK_ADMIN=admin -e KEYCLOAK_ADMIN_PASSWORD="admin" -e KC_CLI_PASSWORD="admin" quay.io/keycloak/keycloak:25.0.1 start-dev`

ssh into container then:

`./kcadm.sh config credentials --server http://localhost:8080 --realm master --user admin
`

`./kcadm.sh get users`

### Anything else?

passing --password was working fine in version 22

**Comment by shawkins** [comment]
> When running the command bellow, we're getting Invalid user credentials [invalid_grant]
kcadm.sh config credentials --server "$KEYCLOAK_SERVER_URL" --realm "master" --user "$KEYCLOAK_ADMIN" --password "$KEYCLOAK_ADMIN_PASSWORD"

This is not reproducible for me. The password works via the env or as a parameter for the credentials or other commands.

The issue with asking for secret will be addressed by #30892

**Comment by taylorfssaraiva** [comment]
you're right @shawkins, the only issue is related to secret not being reused by other commands
